PryntTrimmerView/Classes/Trimmer/PryntTrimmerView.swift

```diff
@@ -172,214 +258,253 @@ public protocol TrimmerViewDelegate: AVAssetTimeSelectorDelegate {
         guard let view = gestureRecognizer.view, let superView = gestureRecognizer.view?.superview else { return }
         let triggeredHandle: TriggeredHandle = view == leftHandleView ? .left : view == rightHandleView ? .right : .unknown
         switch gestureRecognizer.state {
 
         case .began:
+            changeHandleStateColor(color: pressedMainColor)
             if view == leftHandleView {
-                currentLeftConstraint = leftConstraint!.constant
+                currentLeftHandleConstraint = leftHandleConstraint!.constant
+                leftHandleLabel.isHidden = false
             } else if view == rightHandleView {
-                currentRightConstraint = rightConstraint!.constant
+                currentRightHandleConstraint = rightHandleConstraint!.constant
+                rightHandleLabel.isHidden = false
             } else {
-                currentLeftConstraint = leftConstraint!.constant
-                currentRightConstraint = rightConstraint!.constant
+                currentLeftHandleConstraint = leftHandleConstraint!.constant
+                currentRightHandleConstraint = rightHandleConstraint!.constant
+                
+                leftHandleLabel.isHidden = false
+                rightHandleLabel.isHidden = false
             }
-            updateSelectedTime(stoppedMoving: false, triggeredHandle: triggeredHandle)
+            updateSelectedPositionTime(stoppedMoving: false, triggeredHandle: triggeredHandle)
         case .changed:
             let translation = gestureRecognizer.translation(in: superView)
             if view == leftHandleView {
                 updateLeftConstraint(with: translation)
             } else if view == rightHandleView {
                 updateRightConstraint(with: translation)
+                
             } else {
                 updateLeftConstraint(with: translation)
                 updateRightConstraint(with: translation)
             }
-            fixHandlesLabelsPositionIfNeeded()
             if let startTime = startTime, view == leftHandleView {
                 seek(to: startTime)
             } else if let endTime = endTime {
                 seek(to: endTime)
             }
-            updateSelectedTime(stoppedMoving: false, triggeredHandle: triggeredHandle)
-
+            trimmerDelegate?.trimmerHandleDidMove(triggerHandle:triggeredHandle)
+            updateSelectedPositionTime(stoppedMoving: false, triggeredHandle: triggeredHandle)
         case .cancelled, .ended, .failed:
-            updateSelectedTime(stoppedMoving: true, triggeredHandle: triggeredHandle)
+            updateSelectedPositionTime(stoppedMoving: true, triggeredHandle: triggeredHandle)
+            leftHandleLabel.isHidden = true
+            rightHandleLabel.isHidden = true
+            changeHandleStateColor(color: mainColor)
         default: break
         }
     }
-
+    
+    public func updateWidth(increment: CGFloat) {
```

Comment:
as suggested, not please explain / rename widthIncrement

Reply:
removed